contracts/QuestFactory.sol

```diff
@@ -529,16 +570,28 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         ++currentQuest.numberMinted;
         questContract_.singleClaim(msg.sender);
 
-        if(mintFee > 0) processMintFee();
+        if(mintFee > 0) processMintFee(ref_);
 
         emit QuestClaimed(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.rewardAmountInWei());
+
+        if(ref_ != address(0)) {
+            emit QuestClaimedReferred(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.rewardAmountInWei(), ref_, referralFee);
```

Comment:
Same question here that I asked earlier: In the case that this does get emitted, do we want to emit both `QuestClaimed` and `QuestClaimedReferred`?